File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

	static boolean[][] data;
	static boolean[] visited;
	static int N, M, V;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		// 정점개수, 간선개수, 탐색시작정점번호 순으로 입력
		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		V = Integer.parseInt(st.nextToken());

		data = new boolean[N + 1][N + 1];
		visited = new boolean[N + 1];

		for (int i = 0; i < M; i++) {
			st = new StringTokenizer(br.readLine());
			int n = Integer.parseInt(st.nextToken());
			int m = Integer.parseInt(st.nextToken());
			data[n][m] = true;
			data[m][n] = true;
		}

		// DFS
		visited[V] = true;
		DFS(V);

		// BFS
		System.out.println();
		visited = new boolean[N + 1];
		BFS();
	}

	// 깊이 우선 탐색
	public static void DFS(int node) {
		// base case
		System.out.print(node + " ");

		// recursive case
		for (int next = 1; next <= N; next++) {
			if (visited[next] == false && data[node][next] == true && data[next][node] == true) {
				visited[next] = true;
				DFS(next);
			}

		}

	}

	// 너비 우선 탐색
	public static void BFS() {
		Queue<Integer> q = new LinkedList<>();
		q.offer(V);
		visited[V] = true;

		while (!q.isEmpty()) {
			int node = q.poll();
			System.out.print(node + " ");
			
			for (int next = 1; next <= N; next++) {
				if (data[node][next] == true && data[next][node] == true && visited[next] == false) {
					q.offer(next);
					visited[next] = true;
				}
			}
		}
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <string>
#include <queue>
using namespace std;

// DFS, BFS 함수 프로토타입
void DFS(int node);
void BFS(int node);

// 전역 변수로 선언
int N, M, V;
vector<vector<int>> arr;
vector<bool> visited;
string output;

int main()
{
	cin >> N >> M >> V;

	// 2차원 벡터 크기 설정
	arr.resize(N + 1, vector<int>(N + 1, 0)); // 기본값 0으로 초기화
	visited.resize(N + 1, false); // false로 초기화

	for (int i = 0; i < M; i++)
	{
		int n1, n2;
		cin >> n1 >> n2;

		// 양방향 연결 표시
		arr[n1][n2] = 1;
		arr[n2][n1] = 1;
	}

	DFS(V);

	output += "\n"; // 출력 줄바꿈
	visited.assign(N + 1, false); // 사용한 visited 벡터 초기화

	BFS(V);

	cout << output;
}

void DFS(int node)
{
	visited[node] = true;
	output += to_string(node) + " ";


	for (int i = 1; i <= N; i++)
	{
		if (arr[node][i] == 1 && !visited[i]) DFS(i);
	}
}

void BFS(int node)
{
	queue<int> q;
	q.push(node);
	visited[node] = true;

	while (!q.empty())
	{
		node = q.front();
		q.pop();
		output += to_string(node) + " ";

		for (int i = 1; i <= N;i++)
		{
			if (arr[node][i] == 1 && !visited[i])
			{
				q.push(i);
				visited[i] = true;
			}
		}
	}
}